### PR TITLE
Backend1

### LINKED ISSUE #10: Backend: Auth + handling data

### PR DESCRIPTION
**Pull request by water0300**
#10, about halfway done

TL:DR - Implemented Firebase for the mockup rather than Docker/Postgre, implemented a "load from DB to RTF" feature

Note: I had some dep. issues when I launched and followed https://github.com/jpuri/react-draft-wysiwyg/issues/390 which resolved it, just in case you guys have the same prob

Updates:
- Changed RichTextEditor.js to a component, since we need access to createRef()
- Adjusted App render function such that the Route accommodates the Ref
- Established Firebase backend (possibly temporary, permanent if lazy) to store plain text data. Currently using a firebase to store the data
- Made a button that queries the DB and put whatever's in it into the RTF 

To elaborate on Firebase:
I'm doing firebase for now cause after reviewing our specs as well as our interim requirements (i.e. our mock up), I chose this because
- Firebase is relatively easy to set up, and with the interim coming up I figured its the fastest way to get something going
- Firebase is basically a free cloud so no need to buy/download anything, 
I created a temporary database to test things out, I'll have to invite you guys but I think @Shroooom has authorization to add you guys too
Also note, IDK how well this is gonna hold up in terms of saving full documents but for a mock up I reckon its fine

Quick Start for our Firebase DB:
- Once you accept the invite and go to the website, the current data is stored as a "firestore database" (select on the left panel). DB name = "user_data", fields = "UID" and "Data" (case sensitive). For now, I made it so everyone has free R/W permissions, but will change with authorization
- In firebase.js, it just establishes the connection to the DB (settings are on the firebase website under project settings (top left))

To Do:
- RTF to Database
- Authentication

### CODE REVIEW

**Comment by LouisLu705** [comment]
We can make a new pull request for the next part. We need to get what we have one one page for sprint 6 so I am merging this.

Note you need to do a npm install firebase for the new code here to work.
